mgmt/utils/MgmtMarshall.cc

```diff
@@ -37,22 +39,67 @@ union MgmtMarshallAnyPtr {
 
 static char *empty = const_cast<char *>("");
 
+/// Each header is 32 bits. The leading 8 indicate the marshalled type and the last 24 indicate the length.
+static constexpr size_t MGMT_LEN_BITS = 24;
+static constexpr size_t MGMT_LEN_MASK = 0x00FFFFFF;
+
+// For integral values, we ignore the length so we can create constants for the headers.
+static const uint32_t MGMT_INT_HDR  = (MGMT_INT_TYPE << MGMT_LEN_BITS);
+static const uint32_t MGMT_LONG_HDR = (MGMT_LONG_TYPE << MGMT_LEN_BITS);
+
+static uint32_t
+get_type_from_hdr(const MgmtMarshallHdr hdr)
+{
+  return (hdr >> MGMT_LEN_BITS); // shift away the length bits.
```

Comment:
`return` is not a function.

Reply:
if you believe hard enough it is. removed parentheses